core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/IcebergDocument.scala

```diff
@@ -0,0 +1,150 @@
+package edu.uci.ics.amber.core.storage.result.iceberg
+
+import edu.uci.ics.amber.core.storage.model.{BufferedItemWriter, VirtualDocument}
+import edu.uci.ics.amber.core.storage.util.StorageUtil.{withLock, withReadLock, withWriteLock}
+import edu.uci.ics.amber.util.IcebergUtil
+import org.apache.iceberg.{Snapshot, Table}
+import org.apache.iceberg.catalog.{Catalog, TableIdentifier}
+import org.apache.iceberg.data.{IcebergGenerics, Record}
+import org.apache.iceberg.exceptions.NoSuchTableException
+import org.apache.iceberg.io.CloseableIterable
+
+import java.net.URI
+import java.util.concurrent.locks.{ReentrantLock, ReentrantReadWriteLock}
+import scala.jdk.CollectionConverters._
+
+class IcebergDocument[T >: Null <: AnyRef](
+    val catalog: Catalog,
```

Comment:
I think the `catalog` should be created/retrieved inside `IcebergDocument` to make it self-contained.

Reply:
Changed. I added a singleton catalog instance and all IcebergDocuments will use that instance as the catalog.